lms/services/vitalsource.py

```diff
@@ -0,0 +1,60 @@
+import oauthlib
+from oauthlib.oauth1 import SIGNATURE_HMAC_SHA1, SIGNATURE_TYPE_BODY
+
+
+class VitalSourceService:
+    def __init__(self, _context, request):
+        self._oauth_key = request.registry.settings["vitalsource_launch_key"]
+        self._oauth_secret = request.registry.settings["vitalsource_launch_secret"]
+
+    def get_launch_params(self, book_id, cfi, lti_user):
```

Comment:
The code was refactored now so that the `request` is no longer passed in, although I suppose it would be possible to have the `factory` function pass in the LTI user. Is there a good reason not to pass it via `get_launch_params` though?